### PR TITLE
Obmin 295 creating an input with formik field logic

### PR DESCRIPTION
**Pull request by ASGladchenko**
- fixed styles in show-password image centering without variable scrolling
- change structure of Input for same reason 
- create an "InputField" component, for the input to work with the Formik logic( wrapper with logic)
- connected yup library 
- connected InputField in components/index

### CODE REVIEW

File: src/components/input-field/index.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { Field } from 'formik';
+
+import { Input } from '../input';
+import { InputProps } from '../input/types';
+import { fieldProps, InputOnChangeEventType } from './types';
+
+export const InputField = ({ name, label, ...propsField }: InputProps) => {
+  return (
+    <Field name={name}>
+      {({ field: { value, onChange, ...props }, meta, form }: fieldProps) => {
```
Comment (Wolshebnik) [inline src/components/input-field/index.tsx:10]: correct name is fieldProps

File: src/components/input-field/index.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          form.setFieldTouched(name);
+        };
+
+        return (
+          <Input
+            label={label}
+            error={error}
+            onChange={(e) => change(typeof e === 'string' ? e : e.target.value)}
+            {...propsField}
+            {...props}
```
Comment (Wolshebnik) [inline src/components/input-field/index.tsx:23]: fieldProps must be first
and props must be last

File: src/components/input-field/types.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+export type InputOnChangeEventType =
+  | React.ChangeEvent<HTMLInputElement>
+  | string;
+
+export interface IInitialValues {
+  text: string;
+  password: string;
+  tel: string;
+  error?: string;
```
Comment (Wolshebnik) [inline src/components/input-field/types.ts:12]: sort

**Review by Wolshebnik — CHANGES_REQUESTED** [review]

File: src/components/input/styles.ts [hunk 4]
```diff
@@ -146,10 +150,12 @@ export const Input = styled.input<IInput>`
...
-  top: 12px;
+  top: 50%;
   right: 12px;
+  display: flex;
   cursor: pointer;
   transition: 0.33s;
+  transform: translateY(-50%);
 
   svg {
     width: 24px;
```
Comment (Wolshebnik) [inline src/components/input/styles.ts:161]: And I guess You forgot to export your component from storybook
components/index.ts

**Review by Wolshebnik — CHANGES_REQUESTED** [review]

**Review by Kyslytsia — APPROVED** [review]

**Review by Mish-hash — APPROVED** [review]

**Review by Wolshebnik — APPROVED** [review]

**Pull request merged**